src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [Object.assign({}, state)];
+
+  actions.forEach((action) => {
+    let newState = Object.assign({}, result[result.length - 1]);
+
+    switch (action.type) {
+      case 'addProperties':
+        newState = Object.assign({}, newState, action.extraData);
```

Comment:
```suggestion
        newState = Object.assign(newState, action.extraData);
```

Also you can use destructuring of object properties